File: backend/src/main/java/com/example/backend/dto/AuthResponse.java
package com.example.backend.dto;

import com.example.backend.entity.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    private Long userId;
    private String email;
    private Role role;
    private String firstName;
    private String lastName;
    private Boolean isVerified;
    private String message;
}

File: frontend/lib/models/auth_response.dart
import 'user.dart';
import 'role.dart';
import 'verification_status.dart';

class AuthResponse {
  final int userId;
  final String email;
  final Role? role;
  final String firstName;
  final String lastName;
  final bool? isVerified;
  final VerificationStatus verificationStatus;
  final String message;

  AuthResponse({
    required this.userId,
    required this.email,
    this.role,
    required this.firstName,
    required this.lastName,
    this.isVerified,
    this.verificationStatus = VerificationStatus.notSubmitted,
    required this.message,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      userId: json['userId'] ?? 0,
      email: json['email'] ?? '',
      role: Role.fromString(json['role']?.toString()),
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      isVerified: json['isVerified'],
      verificationStatus: verificationStatusFromString(json['verificationStatus']?.toString()),
      message: json['message'] ?? '',
    );
  }

  User toUser() {
    return User(
      id: userId,
      email: email,
      firstName: firstName,
      lastName: lastName,
      role: role,
      isVerified: isVerified,
      verificationStatus: verificationStatus,
    );
  }
}
